### PR TITLE
add hippomocks recipe

### PR DESCRIPTION
**Pull request by tomas-krupa**
Specify library name and version: hippomocks/10960d0

Hippomocks is a single-header mocking framework.
Homepage: https://github.com/dascandy/hippomocks
Author: https://github.com/dascandy

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by luismartingil** [comment]
@uilianries What's the next step here, please?

**Comment by tomas-krupa** [comment]
> [CLA assistant check](https://cla-assistant.io/conan-io/conan-center-index?pullRequest=5165)
> Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/conan-io/conan-center-index?pullRequest=5165) before we can accept your contribution.
> You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/conan-io/conan-center-index?pullRequest=5165) it.

**Comment by uilianries** [comment]
Hi @tomas-krupa @luismartingil  Could you please sign the CLA: https://github.com/conan-io/conan-center-index/pull/5165#issuecomment-816503690. Otherwise we can't accept your PR.

**Comment by luismartingil** [comment]
@uilianries Maybe @tomas-krupa needs to sign it too?

File: recipes/hippomocks/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
+from conans import ConanFile, ConanFile, tools
+import os, glob
+
+class HippomocksConan(ConanFile):
+    name = 'hippomocks'
+    version = '10960d0' # This library is not versioned
```
Comment (uilianries) [inline recipes/hippomocks/all/conanfile.py:6]: Please, read and follow: https://github.com/conan-io/conan-center-index/blob/master/docs/faqs.md#what-version-should-packages-use-for-libraries-without-official-releases

Comment (uilianries) [inline recipes/hippomocks/all/conanfile.py:6]:
```suggestion
```
Version should be declared on recipe file

Please, read and follow: master/docs/faqs.md#what-version-should-packages-use-for-libraries-without-official-releases

File: recipes/hippomocks/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    description = 'Single-header mocking framework.'
+    topics = ("conan", "hippomocks", "mock", "framework")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = 'https://github.com/dascandy/hippomocks.git'
+    license = 'LGPL-2.1'
+    settings = None
+    short_paths = True
```
Comment (uilianries) [inline recipes/hippomocks/all/conanfile.py:13]:
```suggestion
```

File: recipes/hippomocks/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        return "source_subfolder"
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = glob.glob("%s-%s*" % (self.name, self.version))[0]
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def build(self):
+        # This is a header-only library so no build step required
+        pass
```
Comment (uilianries) [inline recipes/hippomocks/all/conanfile.py:27]:
```suggestion
```
not used

File: recipes/hippomocks/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+    def package(self):
+        self.copy('*.h', dst='include/hippomocks', src='%s/HippoMocks/' % self._source_subfolder)
```
Comment (uilianries) [inline recipes/hippomocks/all/conanfile.py:30]:
```suggestion
        self.copy('*.h', dst=os.path.join('include', 'hippomocks'), src=os.path.join(self._source_subfolder, 'HippoMocks'))
```

File: recipes/hippomocks/all/test_package/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+
+add_library(conan-hippomocks INTERFACE ${CONAN_LIBS})
+add_library(Hippomocks::Hippomocks ALIAS conan-hippomocks)
+
+add_executable(${PROJECT_NAME} main.cpp)
+target_link_libraries(${PROJECT_NAME} PRIVATE Hippomocks::Hippomocks)
+
+enable_testing()
+
+add_test(NAME ${PROJECT_NAME} COMMAND ${PROJECT_NAME})
```
Comment (uilianries) [inline recipes/hippomocks/all/test_package/CMakeLists.txt:20]:
```suggestion
cmake_minimum_required(VERSION 3.1)
project(test_package LANGUAGES CXX)

include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
conan_basic_setup(TARGETS)

find_package(Hippomocks CONFIG REQUIRED)

add_executable(${PROJECT_NAME} main.cpp)
target_link_libraries(${PROJECT_NAME} PRIVATE Hippomocks::Hippomocks)

```
Keep it simple, don't use CTest, it hides all output.

File: recipes/hippomocks/all/test_package/conanfile.py [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+    no_copy_source = True
+
+    def build(self):
+        env_build = RunEnvironment(self)
+        with tools.environment_append(env_build.vars):
+            cmake = CMake(self)
+            cmake.configure()
+            cmake.build()
+            if tools.get_env("CONAN_RUN_TESTS", True):
+                cmake.test()
```
Comment (uilianries) [inline recipes/hippomocks/all/test_package/conanfile.py:16]:
```suggestion
        cmake = CMake(self)
        cmake.configure()
        cmake.build() 
```

File: recipes/hippomocks/all/test_package/conanfile.py [hunk 7]
```diff
@@ -0,0 +1,21 @@
+from conans import ConanFile, CMake, tools, RunEnvironment
+import os
+
+class TestPackageConan(ConanFile):
+    settings = "os", "compiler", "build_type", "arch"
+    generators = "cmake"
+    no_copy_source = True
```
Comment (uilianries) [inline recipes/hippomocks/all/test_package/conanfile.py:7]:
```suggestion
```

Not required

File: recipes/hippomocks/config.yml [hunk 8]
```diff
@@ -0,0 +1,3 @@
+versions:
+  "10960d0":
```
Comment (uilianries) [inline recipes/hippomocks/config.yml:2]: Commit versions are not allowed, read CCI FAQs, the commit date must be used.

**Comment by uilianries** [comment]
> Maybe @tomas-krupa needs to sign it too?

Only commit author needs to sign.

**Comment by tomas-krupa** [comment]
> Hi @tomas-krupa @luismartingil Could you please sign the CLA: [#5165 (comment)](https://github.com/conan-io/conan-center-index/pull/5165#issuecomment-816503690). Otherwise we can't accept your PR.

I signed it about a hundred times. Is there any issue with my account?
signed

File: recipes/hippomocks/all/conanfile.py (hunk 1 above)
Comment (tomas-krupa) [reply recipes/hippomocks/all/conanfile.py:6]: thanks a lot, I missed this info

**Comment by uilianries** [comment]
> > Hi @tomas-krupa @luismartingil Could you please sign the CLA: [#5165 (comment)](https://github.com/conan-io/conan-center-index/pull/5165#issuecomment-816503690). Otherwise we can't accept your PR.
> 
> I signed it about a hundred times. Is there any issue with my account?
> signed

You are using 2 emails, take a look on CLA bot comment: https://github.com/conan-io/conan-center-index/pull/5165#issuecomment-816503690

Both must accept the CLA

**Comment by tomas-krupa** [comment]
Im really sorry for the issues with my key, created a new PR at  https://github.com/conan-io/conan-center-index/pull/5254

**Pull request closed without merging**